cmd/gbridge/main.go

```diff
@@ -165,6 +166,35 @@ func main() {
 		Info: proto.DeviceInfo{
 			HwVersion: "1.0",
 		},
+		}); err != nil {
+		log.Fatal(err)
+	}
+
+	// register devices
+	if err := smartHome.RegisterDevice("pborges", gbridge.BasicDevice{
+		Id: "1234567891",
+		Name: proto.DeviceName{
+			Name: "Blind1",
+		},
+		Type: proto.DeviceTypeBlinds,
+		Traits: []gbridge.Trait{
+			gbridge.OpenCloseTrait{
+				DiscreteOnlyOpenClose: true,
+				OpenDirection: []gbridge.OpenCloseTraitDirection{gbridge.OpenCloseTraitDirectionUp, gbridge.OpenCloseTraitDirectionDown},
+				QueryOnlyOpenClose: false,
+				OnExecuteChange: func(ctx gbridge.Context, params interface{}) proto.DeviceError {
+					log.Println("Percent of", ctx.Target.DeviceName(), "should be set to", params)
+					return nil
+				},
+				OnStateHandler: func(ctx gbridge.Context) (gbridge.OpenState, proto.ErrorCode) {
+					log.Println("query state of", ctx.Target.DeviceName())
+					curOpenState := gbridge.OpenState{100.0,gbridge.OpenCloseTraitDirectionUp}
+					return curOpenState, nil
```

Comment:
once the ```OnStateHandler``` changes go in you can ```return []OpenState{curOpenState}``` here, by SUPPORTING multiple directions we are "future proof" if someone decides they want to try and implement it.... ours is not to judge, but to facilitate :P

Reply:
good call